lib/src/config_delayed_merge.cc

```diff
@@ -23,8 +23,8 @@ namespace hocon {
         throw config_exception("called value_type() on value with unresolved substitutions, need to config#resolve() first, see API docs");
     }
 
-    vector<shared_value> const& config_delayed_merge::unmerged_values() const {
-        return _stack;
+    shared_ptr<vector<shared_value>> config_delayed_merge::unmerged_values() const {
+        return make_shared<vector<shared_value>>(_stack);
```

Comment:
Actually, since we're immediately copying the elements into another vector, let's look at returning by value and using move_iterators to move the elements. http://en.cppreference.com/w/cpp/iterator/move_iterator